File: seungyeop/22_12_13_02.java
class Solution {    //다른 사람 풀이 중 좋았던 것: newBoard 길이를 +2 씩 늘리고 제외 범위 없이 입력 후 기존 범위만 카운트
    public int solution(int[][] board) {
        int answer = 0;
        int n = board.length;
        int[][] newBoard = new int[n][n];    //board와 똑같은 크기의 newBoard 생성
        for (int i = 0; i < n; i++) {
            for (int j = 0; j < n; j++) {
                if (board[i][j] == 1) {     //board의 값이 1이라면 newBoard 주변 값에 1을 넣음
                    for (int k = i - 1; k <= i + 1; k++) {
                         for (int l = j - 1; l <= j + 1; l++) {
                            if (k > -1 && k < n && l > -1 && l < n) {   //board 범위를 벗어나는 경우 제외
                                newBoard[k][l] = 1;
                            }
                        }
                    }
                }
            }
        }
        for (int i = 0; i < n; i++) {
            for (int j = 0; j < n; j++) {
                if (newBoard[i][j] == 0) {      //안전지대 개수 카운트
                    answer += 1;
                }
            }
        }
        return answer;
    }
}

File: seungyoon/22_12_13_02.py
def solution(board):
    for i in range(len(board)):
        for j in range(len(board)):
            board_시작 = 0
            board_끝 = len(board) - 1
            위 = i - 1
            아래 = i + 1
            왼쪽 = j - 1
            오른쪽 = j + 1
            위험지대 = 2
            if board[i][j] == 1:
                if 위 >= board_시작:                           # 위
                    if board[위][j] != 1:
                        board[위][j] = 위험지대
                if 아래 <= board_끝:                           # 아래
                    if board[아래][j] != 1:
                        board[아래][j] = 위험지대        
                if 왼쪽 >= board_시작:                          # 왼쪽
                    if board[i][왼쪽] != 1:
                        board[i][왼쪽] = 위험지대
                if 오른쪽 <= board_끝:                          # 오른쪽
                     if board[i][오른쪽] != 1:
                        board[i][오른쪽] = 위험지대
                if 위 >= board_시작 and 왼쪽 >= board_시작:      # 왼쪽 위
                     if board[위][왼쪽] != 1:
                        board[위][왼쪽] = 위험지대
                if 위 >= board_시작 and 오른쪽 <= board_끝:      # 오른쪽 위
                    if board[위][오른쪽] != 1:
                        board[위][오른쪽] = 위험지대
                if 아래 <= board_끝 and 왼쪽 >= board_시작 :     # 왼쪽 아래
                    if board[아래][왼쪽] != 1:
                        board[아래][왼쪽] = 위험지대
                if 아래 <= board_끝 and 오른쪽 <= board_끝 :     # 오른쪽 아래
                    if board[아래][오른쪽] != 1:
                        board[아래][오른쪽] = 위험지대
                        
    # 0의 개수 세기
    answer = 0
    for i in range(len(board)):
        for j in range(len(board)):
            if board[i][j] == 0:
                answer += 1
    return answer
